### ISSUE TITLE
CUDA 11.6 generates invalid device code for all physicalBoundaryFunctors

### ISSUE THREAD

**Issue by BenWibking** [quokka-astro/quokka#21]
nvcc 11.6.55 generates invalid device code for all test problems without fully-periodic boundary conditions:

```
Starting program: /avatar/bwibking/quokka/build/src/HydroShocktube/test_hydro_shocktube shocktube.in
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
warning: Cannot parse .gnu_debugdata section; LZMA support was disabled at compile time
warning: Cannot parse .gnu_debugdata section; LZMA support was disabled at compile time
[Detaching after fork from child process 231811]
[New Thread 0x7ffff4661000 (LWP 231815)]
[New Thread 0x7ffff3e60000 (LWP 231816)]
Initializing CUDA...
[Detaching after fork from child process 231818]
[New Thread 0x7ffff162f000 (LWP 231828)]
[New Thread 0x7ffff0e2e000 (LWP 231829)]
CUDA initialized with 1 GPU per MPI rank; 1 GPU(s) used in total
MPI initialized with 1 MPI processes
MPI initialized with thread support level 0
AMReX (22.01-1-ga73ce6759c91) initialized
warning: Cuda API error detected: cudaLaunchKernel returned (0x62)

warning: Cuda API error detected: cudaGetLastError returned (0x62)

terminate called after throwing an instance of 'std::runtime_error'
  what():  GPU last error detected in file /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_GpuLaunchFunctsG.H line 862: invalid device function

Thread 1 "test_hydro_shoc" received signal SIGABRT, Aborted.
0x00007ffff63ae37f in raise () from /lib64/libc.so.6
(cuda-gdb) bt
#0  0x00007ffff63ae37f in raise () from /lib64/libc.so.6
#1  0x00007ffff6398db5 in abort () from /lib64/libc.so.6
#2  0x00007ffff6d6609b in ?? () from /lib64/libstdc++.so.6
#3  0x00007ffff6d6c53c in ?? () from /lib64/libstdc++.so.6
#4  0x00007ffff6d6b559 in ?? () from /lib64/libstdc++.so.6
#5  0x00007ffff6d6bed8 in __gxx_personality_v0 () from /lib64/libstdc++.so.6
#6  0x00007ffff674cb03 in ?? () from /lib64/libgcc_s.so.1
#7  0x00007ffff674d071 in _Unwind_RaiseException () from /lib64/libgcc_s.so.1
#8  0x00007ffff6d6c7eb in __cxa_throw () from /lib64/libstdc++.so.6
#9  0x0000000000484c22 in amrex::Abort_host (
    msg=0xec5c050 "GPU last error detected in file /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_GpuLaunchFunctsG.H line 862: invalid device function") at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX.cpp:237
#10 0x0000000000484a27 in amrex::Abort (
    msg=0xec5c050 "GPU last error detected in file /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_GpuLaunchFunctsG.H line 862: invalid device function") at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX.H:154
#11 amrex::Abort (msg=...) at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX.cpp:197
#12 0x000000000041a394 in amrex::Gpu::ErrorCheck (
    file=0x81c988 "/avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_GpuLaunchFunctsG.H", line=862)
    at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_GpuError.H:57
#13 0x0000000000453e94 in amrex::ParallelFor<__nv_dl_wrapper_t<__nv_dl_tag<void (amrex::GpuBndryFuncFab<setBoundaryFunctor<ShocktubeProblem> >::*)(amrex::Box const&, amrex::FArrayBox&, int, int, amrex::Geometry const&, double, amrex::Vector<amrex::BCRec, std::allocator<amrex::BCRec> > const&, int, int, amrex::FilccCell&&), &amrex::GpuBndryFuncFab<setBoundaryFunctor<ShocktubeProblem> >::ccfcdoit<amrex::FilccCell>, 1u>, amrex::FilccCell, amrex::Array4<double> const, int const, int const, amrex::Box const, amrex::BCRec*, setBoundaryFunctor<ShocktubeProblem> const, amrex::GeometryData const, double const, int const> > (box=..., f=...)
    at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_GpuLaunchFunctsG.H:862
#14 0x000000000044632b in amrex::ParallelFor<__nv_dl_wrapper_t<__nv_dl_tag<void (amrex::GpuBndryFuncFab<setBoundaryFunctor<ShocktubeProblem> >::*)(amrex::Box const&, amrex::FArrayBox&, int, int, amrex::Geometry const&, double, amrex::Vector<amrex::BCRec, std::allocator<amrex::BCRec> > const&, int, int, amrex::FilccCell&&), &amrex::GpuBndryFuncFab<setBoundaryFunctor<ShocktubeProblem> >::ccfcdoit<amrex::FilccCell>, 1u>, amrex::FilccCell, amrex::Array4<double> const, int const, int const, amrex::Box const, amrex::BCRec*, setBoundaryFunctor<ShocktubeProblem> const, amrex::GeometryData const, double const, int const> > (box=..., f=...)
    at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_GpuLaunchFunctsG.H:1275
#15 0x000000000042864d in amrex::GpuBndryFuncFab<setBoundaryFunctor<ShocktubeProblem> >::ccfcdoit<amrex::FilccCell> (
    this=0x7fffffff8130, bx=..., dest=..., dcomp=0, numcomp=9, geom=..., time=0, bcr=..., bcomp=0, orig_comp=0,
    fillfunc=...) at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_PhysBCFunct.H:383
#16 0x0000000000455a0d in amrex::GpuBndryFuncFab<setBoundaryFunctor<ShocktubeProblem> >::operator() (
    this=0x7fffffff8130, bx=..., dest=..., dcomp=0, numcomp=9, geom=..., time=0, bcr=..., bcomp=0, orig_comp=0)
    at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_PhysBCFunct.H:204
#17 0x0000000000449b91 in amrex::PhysBCFunct<amrex::GpuBndryFuncFab<setBoundaryFunctor<ShocktubeProblem> > >::operator() (this=0x7fffffff80c0, mf=..., icomp=0, ncomp=9, nghost=..., time=0, bccomp=0)
    at /avatar/bwibking/quokka/extern/amrex/Src/Base/AMReX_PhysBCFunct.H:177
#18 0x000000000046cca1 in AMRSimulation<ShocktubeProblem>::fillBoundaryConditions (this=0x7fffffff8620, S_filled=...,
    state=..., lev=0, time=0) at /avatar/bwibking/quokka/src/simulation.hpp:695
#19 0x000000000046a6e1 in AMRSimulation<ShocktubeProblem>::MakeNewLevelFromScratch (this=0x7fffffff8620, level=0,
    time=0, ba=..., dm=...) at /avatar/bwibking/quokka/src/simulation.hpp:628
#20 0x00000000007a0435 in amrex::AmrMesh::MakeNewGrids (this=0x7fffffff8620, time=0)
    at /avatar/bwibking/quokka/extern/amrex/Src/AmrCore/AMReX_AmrMesh.cpp:779
#21 0x00000000006fb236 in amrex::AmrCore::InitFromScratch (this=0x7fffffff8620, time=0)
    at /avatar/bwibking/quokka/extern/amrex/Src/AmrCore/AMReX_AmrCore.cpp:64
#22 0x000000000041e227 in AMRSimulation<ShocktubeProblem>::setInitialConditions (this=0x7fffffff8620)
    at /avatar/bwibking/quokka/src/simulation.hpp:280
#23 0x0000000000410ee8 in problem_main () at /avatar/bwibking/quokka/src/HydroShocktube/test_hydro_shocktube.cpp:332
#24 0x000000000040dde5 in main (argc=2, argv=0x7fffffff8d28) at /avatar/bwibking/quokka/src/main.cpp:41
```

Workaround: Use CUDA <= 11.5.

**Comment by BenWibking** [comment]
It affects all tests without fully-periodic boundary conditions:

```
Test project /avatar/bwibking/quokka/build
      Start  1: ScalarAdvection
 1/18 Test  #1: ScalarAdvection ..................   Passed   59.88 sec
      Start  2: ScalarAdvectionSemiEllipse
 2/18 Test  #2: ScalarAdvectionSemiEllipse .......   Passed   61.43 sec
      Start  3: HydroContact
 3/18 Test  #3: HydroContact .....................   Passed    8.34 sec
      Start  4: HydroLeblanc
 4/18 Test  #4: HydroLeblanc .....................Subprocess aborted***Exception:   1.96 sec
      Start  5: HydroShocktube
 5/18 Test  #5: HydroShocktube ...................Subprocess aborted***Exception:   3.01 sec
      Start  6: ShuOsher
 6/18 Test  #6: ShuOsher .........................Subprocess aborted***Exception:   3.19 sec
      Start  7: HydroSlowMovingShock
 7/18 Test  #7: HydroSlowMovingShock .............Subprocess aborted***Exception:   3.45 sec
      Start  8: HydroVacuum
 8/18 Test  #8: HydroVacuum ......................Subprocess aborted***Exception:   3.11 sec
      Start  9: HydroWave
 9/18 Test  #9: HydroWave ........................   Passed   21.05 sec
      Start 10: RadForce
10/18 Test #10: RadForce .........................Subprocess aborted***Exception:   3.33 sec
      Start 11: MarshakWave
11/18 Test #11: MarshakWave ......................Subprocess aborted***Exception:   3.20 sec
      Start 12: MarshakWaveCGS
12/18 Test #12: MarshakWaveCGS ...................Subprocess aborted***Exception:   3.06 sec
      Start 13: RadiationPulse
13/18 Test #13: RadiationPulse ...................Subprocess aborted***Exception:   3.28 sec
      Start 14: RadiationStreaming
14/18 Test #14: RadiationStreaming ...............Subprocess aborted***Exception:   3.06 sec
      Start 15: SuOlsonTest
15/18 Test #15: SuOlsonTest ......................Subprocess aborted***Exception:   2.08 sec
      Start 16: RadiationPressureTube
16/18 Test #16: RadiationPressureTube ............Subprocess aborted***Exception:   2.94 sec
      Start 17: RadhydroShock
17/18 Test #17: RadhydroShock ....................Subprocess aborted***Exception:   2.83 sec
      Start 18: RadhydroShockCGS
18/18 Test #18: RadhydroShockCGS .................Subprocess aborted***Exception:   2.75 sec

22% tests passed, 14 tests failed out of 18

Total Test time (real) = 193.20 sec

The following tests FAILED:
	  4 - HydroLeblanc (Subprocess aborted)
	  5 - HydroShocktube (Subprocess aborted)
	  6 - ShuOsher (Subprocess aborted)
	  7 - HydroSlowMovingShock (Subprocess aborted)
	  8 - HydroVacuum (Subprocess aborted)
	 10 - RadForce (Subprocess aborted)
	 11 - MarshakWave (Subprocess aborted)
	 12 - MarshakWaveCGS (Subprocess aborted)
	 13 - RadiationPulse (Subprocess aborted)
	 14 - RadiationStreaming (Subprocess aborted)
	 15 - SuOlsonTest (Subprocess aborted)
	 16 - RadiationPressureTube (Subprocess aborted)
	 17 - RadhydroShock (Subprocess aborted)
	 18 - RadhydroShockCGS (Subprocess aborted)
```

**Comment by BenWibking** [comment]
Closing since it "will be fixed in the next CUDA minor version release"

**Comment by BenWibking** [comment]
This is fixed in CUDA 11.7.
